lib/serializers/bib.js

```diff
@@ -440,17 +440,21 @@ var fromMarcJson = (object, datasource) => {
     })
 
     // Add various parallel literals:
-    ; ['Parallel title', 'Parallel title display', 'Parallel series statement'].forEach((name) => {
-      fieldMapper.getMapping(name, (fieldMapping) => {
-        fieldMapping.paths.forEach((path) => {
-          const parallelValues = object.parallel(path.isParallelFor, path.subfields)
-          if (parallelValues && parallelValues.length > 0) {
-            parallelValues.forEach((parallelValue) => {
-              const sourceRecordPath = `880[$u^=${path.isParallelFor}] ${path.subfields.map((subfield) => `$${subfield}`).join(' ')}`
-              builder.add(fieldMapping.pred, { literal: parallelValue || ' ' }, null, { path: sourceRecordPath })
-            })
-          }
-        })
+
+    const parallelMappings = Object.values(fieldMapper.data).filter(mapping =>
+      mapping.paths && mapping.paths.some(path => path.isParallelFor)
+    )
+
+    parallelMappings.forEach((fieldMapping) => {
+      fieldMapping.paths.forEach((path) => {
+        const subfields = path.subfields || [];
+        const parallelValues = object.parallel(path.isParallelFor, subfields)
+        if (parallelValues && parallelValues.length > 0) {
+          parallelValues.forEach((parallelValue) => {
+            const sourceRecordPath = `880[$u^=${path.isParallelFor}] ${subfields.map((subfield) => `$${subfield}`).join(' ')}`
+            builder.add(fieldMapping.pred, { literal: parallelValue || ' ' }, null, { path: sourceRecordPath })
+          })
+        }
```

Comment:
Currently this is looping over each of the "paths", calling `builder.add` as values are found. This works fine when there are parallel values. When a given mapping has no parallel values we end up calling `builder.add` on a bunch of empty values, resulting in a bunch of useless statements (or empty statements at the end of the list). I wonder if we should apply the rtrim here (instead of in the SierraRecord.parallel call). We'd have to change the structure of the above slightly to _collect_ the statements to be added, then rtrim them (remove blank statements from the end), and _then_ call `builder.add` on them (if any remain).. Essentially we have to do that cleanup work either here or in the indexer later..